### ISSUE TITLE
UseDotNet@2 cant find sdk version matching 3.x

### ISSUE THREAD

**Issue by justinswork** [microsoft/azure-pipelines-tasks#11569]
## Note  
Issues in this repo are for tracking bugs, feature requests and questions for the tasks in this repo 

For a list:  
https://github.com/Microsoft/azure-pipelines-tasks/tree/master/Tasks

If you have an issue or request for the Azure Pipelines service, use developer community instead:  

https://developercommunity.visualstudio.com/spaces/21/index.html )

## Required Information

Entering this information will route you directly to the right team and expedite traction.

**Question, Bug, or Feature?**  
*Type*: Bug

**Enter Task Name**:UseDotNet@2

https://github.com/Microsoft/azure-pipelines-tasks/tree/master/Tasks

## Environment
- Server - Azure Pipelines

- Agent - Private
Windows 10

## Issue Description

Unable to find a matching version of DotNetCore sdk for 3.x
Including parameter **includePreviewVersions** finds the latest preview version of 3.1, but I want the latest current version. (currently 3.0.100)
```
##[section]Starting: UseDotNet
==============================================================================
Task         : Use dotnet
Description  : Acquires a specific version of the .NET Core SDK from the internet or the local cache and adds it to the PATH. Use this task to change the version of .NET Core used in subsequent tasks. Additionally provides proxy support.
Version      : 2.0.5
Author       : Microsoft Corporation
Help         : https://aka.ms/AA4xgy0
==============================================================================
Tool to install: .NET Core sdk version 3.x.
##[error]Failed while getting version 3.x from channel 3.1 with error: {}
##[error]sdk version matching: 3.x could not be found
##[section]Finishing: UseDotNet
```

### Error logs

```
##[error]Failed while getting version 3.x from channel 3.1 with error: {}
##[error]sdk version matching: 3.x could not be found
```

**Comment by alastairupton** [comment]
I have just changed mine to 3.0.x - I guess work has started on the 3.1 branch so, for stability, it is probably best to use 3.0.x.

**Comment by mhdubose** [comment]
While changing it to 3.0.x works, 3.x should still work as well. This broke my pipelines as well, and the documentation still states that using Major.x will supply the latest Minor version.

**Comment by joshlang** [comment]
Yeah, this broke our build too

**Comment by wrkettlitz** [comment]
Same

**Comment by tejasd1990** [comment]
Hi @justinswork, @alastairupton, @mhdubose, @joshlang, @wrkettlitz
Thanks for reporting this. We are investigating this issue. Will update soon. Meanwhile, as a workaround, can you try with version 2.x ?

**Comment by tejasd1990** [comment]
Hi all,
The issue seems with the latest 3.1 version which is in preview state. The versions are fetched from
[this ](https://github.com/dotnet/core/blob/master/release-notes/releases-index.json) releases file.

Another workaround is that you can copy this file to global.json to your build/release repo, remove the first entry corresponding to _"latest-runtime": "3.1.0-preview1.19506.1"_, and enable the _"Use global json"_ flag in the build/release. Then it should work even with 3.x version

**Comment by tejasd1990** [comment]
Hi all, we are working on the fix and will update once it is rolled out. Meanwhile, you can unblock yourselves with the above suggested workarounds.

**Comment by Eneuman** [comment]
Same, this has broken all our 26 build pipelines!!

Acording to https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/tool/dotnet-core-tool-installer?view=azure-devops, 3.x is a valid value.

Please fix this asap as this is a blocking issue.

**Comment by joshlang** [comment]
@Eneuman There's an easy fix for you in the meantime - change it to 3.0.x

They've already merged a fix so it should be live soon, anyway.

**Comment by tejasd1990** [comment]
Hi all,
we have started the rollout for this fix. I will update once the rollout is complete.

**Comment by tejasd1990** [comment]
Hi all,
the fix has been rolled out. Let us know in case you still face the error. We will close this issue.

**Comment by ezuidema** [comment]
I got to this thread by the issue referenced by @hiyadav, but our build actually broke because it is now using 3.x while we specify it to use 2.2. Is this related?
![image](https://user-images.githubusercontent.com/4758803/67609179-c3436700-f78b-11e9-80c2-bb04945c4a6b.png)

**Comment by glennawatson** [comment]
@ezuidema I would just open a new issue if none are directly related. From experience having multiple slightly related bugs combined into the one issue makes it hard to track.

**Comment by vinodkumar3** [comment]
Closing this issue item as the original issue got fixed. 
@ezuidema feel free to raise new issue item for your scenario!
